src/main/kotlin/de/cyface/collector/storage/GridFsStorageService.kt

```diff
@@ -0,0 +1,311 @@
+    }
+
+    override fun bytesUploaded(uploadIdentifier: UUID): Future<Long> {
+        val ret = Promise.promise<Long>()
+        val temporaryFileName = pathToTemporaryFile(uploadIdentifier).toFile().absolutePath
+        fs.props(temporaryFileName).onSuccess { props: FileProps ->
+            // Wrong chunk uploaded
+            val byteSize = props.size()
+            ret.complete(byteSize)
+        }.onFailure(ret::fail)
+
+        return ret.future()
+    }
+
+    override fun clean(uploadIdentifier: UUID): Future<Void> {
+        val ret = Promise.promise<Void>()
+        val pathToTemporaryFile = pathToTemporaryFile(uploadIdentifier)
+        fs.delete(pathToTemporaryFile.absolutePathString())
+            .onSuccess(ret::complete)
+            .onFailure(ret::fail)
+        return ret.future()
+    }
+
+    override fun startPeriodicCleaningOfTempData(uploadExpirationTime: Long, vertx: Vertx) {
+        // Schedule upload file cleaner task
+        vertx.setPeriodic(uploadExpirationTime) {
+            // Remove deprecated temp files
+            // There is no way to look through all sessions to identify unreferenced files. Thus, we remove files which
+            // have not been changed for a long time. The MeasurementHandler handles sessions with "missing" files.
+            fs.readDir(FILE_UPLOADS_FOLDER.absolutePathString()).onSuccess { uploadFiles ->
+                uploadFiles.filter { pathname ->
+                    val path = Paths.get(pathname)
+                    path.isRegularFile() && path.getLastModifiedTime().toMillis() > uploadExpirationTime
+                }.forEach { pathname ->
+                    LOGGER.debug("Cleaning up temp file: {}", pathname)
+                    fs.delete(pathname).onFailure {
+                        LOGGER.warn("Failed to remove temp file: {}", pathname)
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Finds the storage path on the local file system to the temporary data file, based on the `uploadIdentifier`.
+     */
+    private fun pathToTemporaryFile(uploadIdentifier: UUID): Path {
+        return Paths.get(uploadFolder.path, uploadIdentifier.toString())
+    }
+
+    /**
+     * Stores a [Measurement] to a Mongo database. This method never fails. If a failure occurs it is logged and
+     * status code 422 is used for the response.
+     *
+     * @param measurement The measured data to write to the Mongo database.
+     * @param temporaryStorage The temporary storage for the uploaded data to store.
+     */
+    private fun storeToMongoDB(measurement: Measurement, temporaryStorage: Path): Future<ObjectId> {
+        val promise = Promise.promise<ObjectId>()
+        LOGGER.debug(
+            "Insert measurement with id {}:{}!", measurement.metaData.deviceIdentifier,
+            measurement.metaData.measurementIdentifier
+        )
+
+        val bucketServiceCreationResult = mongoClient.createDefaultGridFsBucketService()
```

Comment:
4 words in one variable gets really hard to read, that's why I decided to leave "result" out (always) ^^